gen3-client/jwt/functions.go

```
@@ -132,6 +139,37 @@ func (f *Functions) ParseFenceURLResponse(resp *http.Response) (JsonMessage, err
 	return msg, nil
 }
 
+func (f *Functions) CheckForShepherdAPI(profile string) (bool, error) {
+	_, res, err := f.GetResponse(profile, "", commonUtils.ShepherdVersionEndpoint, "GET", "", nil)
+	if err != nil {
+		return false, errors.New("Error occurred during generating HTTP request: " + err.Error())
+	}
+	if res.StatusCode != 200 {
+		return false, nil
+	}
+	bodyBytes, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return false, errors.New("Error occurred when reading HTTP request: " + err.Error())
+	}
+	body, err := strconv.Unquote(string(bodyBytes))
+	if err != nil {
+		return false, fmt.Errorf("Error occurred when parsing version from Shepherd: %v: %v", string(body), err)
+	}
+	// Compare the version in the response to the target version
+	ver, err := semver.Make(body)
+	if err != nil {
+		return false, fmt.Errorf("Error occurred when parsing version from Shepherd: %v: %v", string(body), err)
+	}
+	minVer, err := semver.Parse(commonUtils.MinAcceptableShepherdVersion)
+	if err != nil {
+		return false, fmt.Errorf("Error occurred when parsing minimum acceptable Shepherd version: %v: %v", commonUtils.MinAcceptableShepherdVersion, err)
+	}
+	if ver.GTE(minVer) {
+		return true, nil
+	}
+	return false, nil
```

should we return an error in this case? like `version too low blabla...`